locust/test/test_main.py

```diff
@@ -381,22 +699,27 @@ def test_default_headless_spawn_options(self):
                     "DEBUG",
                     "--exit-code-on-error",
                     "0",
-                    # just to test --stop-timeout argument parsing, doesnt actually validate its function:
                     "--stop-timeout",
                     "1s",
                 ],
-                stdout=PIPE,
-                stderr=PIPE,
-                text=True,
-            )
-            stdout, stderr = proc.communicate(timeout=4)
-            self.assertNotIn("Traceback", stderr)
-            self.assertIn('Spawning additional {"UserSubclass": 1} ({"UserSubclass": 0} already running)...', stderr)
-            self.assertEqual(0, proc.returncode)
+            ) as popen_ctx:
+                try:
+                    popen_ctx.process.wait(timeout=4)
+                except subprocess.TimeoutExpired:
+                    print("Process timed out")
```

Comment:
Do you really want to catch this exception? Is it not more clear if you just allow it to crash?